libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/ServerPickerActivity.java

```diff
@@ -300,7 +300,9 @@ public void onAuthConfigFetched() {
          * The only other way to do this is with the NO_HISTORY flag on the custom tab, however
          * this will cause it to always reload on background -- breaking MFA.
          */
-        startActivity(intent);
+        if (!SalesforceSDKManager.getInstance().getIsTestRun()) {
+            startActivity(intent);
+        }
```

Comment:
I really don't love this, but it seems to be the only way to get the tests to run consistently.